python/ray/data/tests/test_dataset.py

```diff
@@ -5416,6 +5416,17 @@ def test_ragged_tensors(ray_start_regular_shared):
     ]
 
 
+def test_iterator(ray_start_regular_shared):
```

Comment:
Not sure if I missed it, but should we also tests for `iter_batches` and `stats`?